src/main/java/edu/kit/compiler/codegen/pattern/UnknownPattern.java

```diff
@@ -0,0 +1,60 @@
+package edu.kit.compiler.codegen.pattern;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import edu.kit.compiler.codegen.MatcherState;
+import edu.kit.compiler.codegen.Operand;
+import edu.kit.compiler.codegen.Util;
+import edu.kit.compiler.intermediate_lang.Instruction;
+import firm.bindings.binding_irnode.ir_opcode;
+import firm.nodes.Node;
+import lombok.RequiredArgsConstructor;
+
+public final class UnknownPattern implements Pattern<InstructionMatch> {
+
+    @Override
+    public InstructionMatch match(Node node, MatcherState matcher) {
+        if (node.getOpCode() == ir_opcode.iro_Unknown) {
+            var size = Util.getSize(node.getMode());
+            var targetRegister = matcher.getNewRegister(size);
+            return new UnknownMatch(node, targetRegister);
+        } else {
+            return InstructionMatch.none();
+        }
+    }
+
+    @RequiredArgsConstructor
+    private static final class UnknownMatch extends InstructionMatch.Basic {
+
+        private final Node node;
+        private final int targetRegister;
+
+        @Override
+        public Node getNode() {
+            return node;
+        }
+
+        @Override
+        public Optional<Integer> getTargetRegister() {
+            return Optional.of(targetRegister);
+        }
+
+        @Override
+        public Stream<Operand> getOperands() {
+            return Stream.empty();
+        }
+
+        @Override
+        public Stream<Node> getPredecessors() {
+            return Stream.empty();
+        }
+
+        @Override
+        public List<Instruction> getInstructions() {
+            return List.of(Instruction.newOp("nop", List.of(),
```

Comment:
Why is it necessary to output a `nop` instruction, instead of returning an empty list?

Reply:
It's a bit of a hack. I need to allocate a register but don't have anything to initialize it with. I could instead write some value to the register (it's UB so it doesn't matter which value).

Reply:
Hm, I see. Could we remove the `nop` at a later point? After all, it shouldn't be necessary to include it in the output file, right?

Reply:
We don't need the nop during execution, so it could be removed once register allocation is complete.

Reply:
Where would be the right place to do so? Should we add peephole optimizations now or is it enough to ignore `nop`s during the assembly file writing?

Reply:
For the assembly writer to remove instructions might be a bit unexpected, a peephole optimization would probably be the cleaner solution.

Reply:
Of course, but implementing this in a clean way would require some framework for peephole optimizations. That seems like an overkill for only removing `nop`s. Maybe we can target this later.

Reply:
You're right, if we want to remove `nop`s we could do it in the assembly writer for now.

Reply:
Or just ignore them for now ;) After all, having some `nop`s doesn't hurt us.